includes/REST_API/Stories_Controller.php

```diff
@@ -270,128 +289,137 @@ public function get_items( $request ) {
-		 *
-		 * @since 1.0.0
-		 *
-		 * @param array           $args    Key value array of query var to query value.
-		 * @param WP_REST_Request $request The request used.
-		 */
-		$args       = apply_filters( "rest_{$this->post_type}_query", $args, $request );
-		$query_args = $this->prepare_items_query( $args, $request );
+			if ( $tax_include ) {
+				$terms            = [];
+				$include_children = false;
+				$operator         = 'IN';
 
-		$taxonomies = wp_list_filter( get_object_taxonomies( $this->post_type, 'objects' ), [ 'show_in_rest' => true ] );
+				if ( rest_is_array( $tax_include ) ) {
+					$terms = $tax_include;
+				} elseif ( rest_is_object( $tax_include ) ) {
+					$terms            = empty( $tax_include['terms'] ) ? [] : $tax_include['terms'];
+					$include_children = ! empty( $tax_include['include_children'] );
 
-		if ( ! empty( $request['tax_relation'] ) ) {
-			$query_args['tax_query'] = [ 'relation' => $request['tax_relation'] ]; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
-		}
+					if ( isset( $tax_include['operator'] ) && 'AND' === $tax_include['operator'] ) {
+						$operator = 'AND';
+					}
+				}
 
-		foreach ( $taxonomies as $taxonomy ) {
-			$base        = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
-			$tax_exclude = $base . '_exclude';
-
-			if ( ! empty( $request[ $base ] ) ) {
-				$query_args['tax_query'][] = [
-					'taxonomy'         => $taxonomy->name,
-					'field'            => 'term_id',
-					'terms'            => $request[ $base ],
-					'include_children' => false,
-				];
+				if ( $terms ) {
+					$args['tax_query'][] = [
+						'taxonomy'         => $taxonomy->name,
+						'field'            => 'term_id',
+						'terms'            => $terms,
+						'include_children' => $include_children,
+						'operator'         => $operator,
+					];
+				}
 			}
 
-			if ( ! empty( $request[ $tax_exclude ] ) ) {
-				$query_args['tax_query'][] = [
-					'taxonomy'         => $taxonomy->name,
-					'field'            => 'term_id',
-					'terms'            => $request[ $tax_exclude ],
-					'include_children' => false,
-					'operator'         => 'NOT IN',
-				];
+			if ( $tax_exclude ) {
+				$terms            = [];
+				$include_children = false;
+
+				if ( rest_is_array( $tax_exclude ) ) {
+					$terms = $tax_exclude;
+				} elseif ( rest_is_object( $tax_exclude ) ) {
+					$terms            = empty( $tax_exclude['terms'] ) ? [] : $tax_exclude['terms'];
+					$include_children = ! empty( $tax_exclude['include_children'] );
+				}
+
+				if ( $terms ) {
+					$args['tax_query'][] = [
+						'taxonomy'         => $taxonomy->name,
+						'field'            => 'term_id',
+						'terms'            => $terms,
+						'include_children' => $include_children,
+						'operator'         => 'NOT IN',
+					];
+				}
 			}
 		}
 
+		return $args;
+	}
+
+	/**
+	 * Add response headers, with post counts.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
```

Comment:
Can't workout how to refactor this method to get this number down. Any ideas?

Reply:
Splitting up into smaller methods is an easy way. But I'd be OK with this being done in a new ticket under the #5066 epic.

Reply:
I did look into refactoring this method. It has a lot of different code paths, because of the permissions model of WordPress. I am not sure how we change this without making the method unreadable later.